src/tests/ll_verification.py

```diff
@@ -723,6 +723,269 @@ def ll_ddi_adv_bv_21_c(transport, upperTester, lowerTester, trace):
+            success = success and 'ADI' in packet.payload
+            success = success and packet.payload['ADI'].SID == Sid
+            for superiorPacket in packet.payload['SuperiorPackets']:
+                success = success and packet.payload['ADI'].DID == superiorPacket.payload['ADI'].DID
+                success = success and packet.ts >= superiorPacket.ts + 300
+            if round.DataLength > 0:
+                if round.DataLength > len(packet.payload['AD']):
+                    success = success and 'AuxPtr' in packet.payload
+                else:
+                    # Check advertising data against input
+                    completeAdvDataFound += 1
+                    success = success and len(packet.payload['AD']) == round.DataLength
+                    for i in range(round.DataLength):
+                        success = success and packet.payload['AD'][i] == advData[i]
+            else:
+                success = 'AD' not in packet.payload
+
+        def collectChainedAdvData(packet):
+            advData = bytes([])
+            if packet.payload['SuperiorPackets'][0].type.name != 'ADV_EXT_IND':
+                advData = collectChainedAdvData(packet.payload['SuperiorPackets'][0])
+            advData += packet.payload['AD']
+            return advData
+
+        # Check AUX_CHAIN_INDs
+        # AdvMode set to 00b; The AUX_CHAIN_IND PDU shall include the ADI field matching the ADI field from AUX_ADV_IND
+        # and the AdvData field containing additional data. The AUX_CHAIN_IND PDU shall not include the AdvA,
+        # TargetA, TxPower, or SyncInfo fields
+        # The time between a PDU containing an AuxPtr field and the PDU to which it refers shall be greater than or equal to T_MAFS
+        for packet in packets.fetch(packet_filter=('AUX_CHAIN_IND')):
+            success = success and packet.payload['AdvMode'] == 0b00
+            success = success and 'AdvA' not in packet.payload
+            success = success and 'TargetA' not in packet.payload
+            success = success and 'TxPower' not in packet.payload
+            success = success and 'SyncInfo' not in packet.payload
+            success = success and 'ADI' in packet.payload
+            success = success and packet.payload['ADI'].SID == Sid
+            for superiorPacket in packet.payload['SuperiorPackets']:
+                success = success and packet.payload['ADI'].DID == superiorPacket.payload['ADI'].DID
+                success = success and packet.ts >= superiorPacket.ts + 300
+            if 'AuxPtr' not in packet.payload:
+                # Collect complete advertising data and check advertising data against input
+                completeAdvDataFound += 1
+                collectedAdvData = collectChainedAdvData(packet)
+                success = success and len(collectedAdvData) == round.DataLength
+                for i in range(round.DataLength):
+                    success = success and collectedAdvData[i] == advData[i]
```

Comment:
Could this be just `success = success and collectedAdvData == advData`?

Reply:
See earlier answer